### PR TITLE
Fix manual vLLM Qwen3 sharding bug when trainer export the weights

### PR DESCRIPTION
**Pull request by JenniferWang**
Thanks to @casteryh 's PR of fixing the integration test https://github.com/meta-pytorch/forge/pull/217, we are able to see the buggy behavior of the manual sharding logic.

This diff basically showcases _why_ the test failed and how easy it is to introduce **silent** data correctness issue if we keep pursuing this route of manual sharding.  

The plan is land these two PRs ASAP
1. Fix the broken test: https://github.com/meta-pytorch/forge/pull/217
2. Replace the manual sharding with vanilla load weight calls: https://github.com/meta-pytorch/forge/pull/184  

At the same time, this PR fixes the existing Qwen3 (non MoE) sharding. 

### Before the fix:
* `test_policy_update` integration test fails when tp > 1
* `grpo` (Qwen 1.7B) loss function super high at the start of the trainer
image
https://meta.wandb.io/jiyue/grpo-training/runs/hh3bht2w?nw=nwuserjiyue

### After the fix:
*  `test_policy_update` integration test passes when tp > 1
* `grpo` (Qwen 1.7B) loss function is much more reasonable??
image
https://meta.wandb.io/jiyue/grpo-training/runs/gnvlw7dc?nw=nwuserjiyue

### CODE REVIEW

**Comment by allenwang28** [comment]
@JenniferWang life pro tip, you should be able to do `pre-commit install` (assuming you've `pip install pre-commit`) which will handle the linter for you automatically every time you do a `git commit`
